### PR TITLE
register serializer & tests

### PR DESCRIPTION
**Pull request by malisara**

### CODE REVIEW

File: word_nerd/users/models.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.db import models
```
Comment (inejc) [inline word_nerd/users/models.py]: You can delete this file if you don't need it

File: word_nerd/users/views.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from rest_framework.generics import CreateAPIView
+
+from .serializers import RegisterSerializer
+
+
+class RegisterUserApi(CreateAPIView):
```
Comment (inejc) [inline word_nerd/users/views.py]: I don't know what naming convention django-rest uses, maybe we can use the same, e.g. `RegisterUserAPIView`?

File: word_nerd/users/serializers.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+from django.contrib.auth.models import User
+import django.contrib.auth.password_validation as django_validators
+from django.core import exceptions
+from rest_framework import serializers
+
+
+class RegisterSerializer(serializers.ModelSerializer):
```
Comment (inejc) [inline word_nerd/users/serializers.py]: NIT: maybe `RegisterUserSerializer`

File: word_nerd/users/serializers.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        model = User
+        fields = ['username', 'password', 'password2']
+
+    def validate(self, data):
+        password = data['password']
+        confirmed_password = data.pop('password2')
+
+        errors = {}
+        if password != confirmed_password:
+            raise serializers.ValidationError(f"Passwords don't match")
```
Comment (inejc) [inline word_nerd/users/serializers.py]: The string doesn't need an `f` here

File: word_nerd/users/serializers.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        try:
```
Comment (inejc) [inline word_nerd/users/serializers.py:20]: NIT: I would define `errors` here instead since this is where you need them. E.g.
```python
if password != confirmed_password:
    raise serializers.ValidationError("Passwords don't match")

errors = {}
try:
...
```

File: word_nerd/users/serializers.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+from django.contrib.auth.models import User
+import django.contrib.auth.password_validation as django_validators
+from django.core import exceptions
```
Comment (inejc) [inline word_nerd/users/serializers.py]: Maybe `as django_exceptions` as `exceptions` is very general

File: word_nerd/users/serializers.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        if password != confirmed_password:
+            raise serializers.ValidationError(f"Passwords don't match")
+        try:
+            # validate the password and catch errors
+            django_validators.validate_password(
+                password=password, user=User(data['username'], data['password']))
+        except exceptions.ValidationError as e:
+            errors['password'] = list(e.messages)
+        if errors:
+            raise serializers.ValidationError(errors)
```
Comment (inejc) [inline word_nerd/users/serializers.py]: This can be simplified, I would simply not use `errors`, i.e.:
```python
except exceptions.ValidationError as e:
    raise serializers.ValidationError({'password': list(e.messages)})
return data
```

File: word_nerd/users/serializers.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        return data
+
+    def create(self, validated_data):
+        try:
+            user = User.objects.create_user(username=validated_data['username'],
+                                            password=validated_data['password'])
+            return user
```
Comment (inejc) [inline word_nerd/users/serializers.py]:
```python
return User.objects. ...
```

File: word_nerd/users/serializers.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        except Exception as e:
+            return e
```
Comment (inejc) [inline word_nerd/users/serializers.py]: It's never good to catch all exceptions as `except Exception as e:`. You need to figure out which specific exception this function `create_user` raises and catch that. Also, You usually never `return` exceptions but rather `raise` them. In this case, if you simply write
```python
except Exception as e:
    raise e
```
this is the same as not using a `try except` statement so you can simply delete it

File: word_nerd/tests/test_register.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        response = self.client.post(self.url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(User.objects.count(), 0)
+        self.assertEqual(
+            response.data['non_field_errors'][0], "Passwords don't match")
+
+    def test_empty_password(self):
+        data = {'username': 'testuser1',
+                'password': 'testnogeslo1', 'password2': ''}
```
Comment (inejc) [inline word_nerd/tests/test_register.py]: I think this test is again testing "passwords don't match" functionality. You need to send `'password': '', 'password2': ''` to test empty password. Also, it would be useful to check the returned error message in this test

File: word_nerd/tests/test_register.py [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_weak_password(self):
+        data = {'username': 'testuser1',
+                'password': 'password', 'password2': 'password'}
+        response = self.client.post(self.url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(User.objects.count(), 0)
+        self.assertEqual(response.data['password']
+                         [0], 'This password is too common.')
```
Comment (inejc) [inline word_nerd/tests/test_register.py]: NIT: this is formatted oddly, can you please fix:
```python
self.assertEqual(response.data['password'][0],
                 'This password is too common.')
```

**Review by inejc — COMMENTED** [review]
Nice! Really like that you started writing tests :)

**Review by inejc — APPROVED** [review]
